### PR TITLE
[gha] Use poetry for tests and releases

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR uses Poetry for the installation of the dependencies and running the tests in the GitHub Actions workflow.

It also adds a new GitHub Action to create the release for Perceval Mozilla. It generates a package, tests it, and publishes the release on PyPi and GitHub whenever a tag is generated and pushed.

It also updates the documentation of the dependencies.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - '*.*.*'
+      - '*.*.*-*'
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build package using Poetry and store result
+        uses: jjmerchante/actions/build@main
```
Comment (vchrombie) [inline .github/workflows/release.yml:15]:
```suggestion
        uses: chaoss/grimoirelab-github-actions/build@master
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          PACKAGE_NAME=`(cd dist && ls *whl | cut -f 1 -d "-")` && echo $PACKAGE_NAME
+          pip install --pre --find-links ./dist/ $PACKAGE_NAME
+          cd tests && python run_tests.py
+
+  release:
+    needs: [tests]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create a new release for the repository
+        uses: jjmerchante/actions/release@main
```
Comment (vchrombie) [inline .github/workflows/release.yml:53]:
```suggestion
        uses: chaoss/grimoirelab-github-actions/release@master
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  publish:
+    needs: [tests]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Publish the package in PyPI
+        uses: jjmerchante/actions/publish@main
```
Comment (vchrombie) [inline .github/workflows/release.yml:62]:
```suggestion
        uses: chaoss/grimoirelab-github-actions/publish@master
```

**Review by vchrombie — CHANGES_REQUESTED** [review]
The PR looks good, only these final changes might be needed since we published the actions to the [chaoss/grimoirelab-github-actions](https://github.com/chaoss/grimoirelab-github-actions) repo.

File: releases/unreleased/use-native-namespace-packages.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+title: Add Poetry support
+category: added
+author: Jose Javier Merchante <jjmerchante@bitergia.com>
+issue: #59
```
Comment (vchrombie) [inline releases/unreleased/use-native-namespace-packages.yml:5]:
```suggestion
issue: 59
```

`#` will be added by the release-tools directly wherever needed.
https://github.com/Bitergia/release-tools/releases/tag/0.3.0

**Review by vchrombie — CHANGES_REQUESTED** [review]
I think it is fine if you don't want to include the changelog file in this PR. I understand the changelog check will fail but I feel we can choose to ignore it for this one time.

I would prefer to start using changelog entries once we publish fresh releases of all the tools. Anyways, I would leave the decision to you, feel free to decide.

I had a small suggestion, rest of the PR looks good to me.

File: releases/unreleased/use-native-namespace-packages.yml (hunk 4 above)
Comment (jjmerchante) [reply releases/unreleased/use-native-namespace-packages.yml]: I decided to remove the file until we decide to use changelog entries with the new releases.

Comment (vchrombie) [reply releases/unreleased/use-native-namespace-packages.yml]: Perfect @jjmerchante :+1:

**Review by vchrombie — APPROVED** [review]
LGTM!

**Comment by vchrombie** [comment]
Ignoring the failed changelog check for now until we decide to use changelog entries with the new releases.

**Comment by vchrombie** [comment]
Merging it

**Pull request merged**